validator/src/main.rs

```diff
@@ -2448,10 +2455,13 @@ pub fn main() {
     accounts_index_config.index_limit_mb =
         if let Some(limit) = value_t!(matches, "accounts_index_memory_limit_mb", usize).ok() {
             IndexLimitMb::Limit(limit)
-        } else if matches.is_present("disable_accounts_disk_index") {
-            IndexLimitMb::InMemOnly
-        } else {
+        } else if matches.is_present("enable_accounts_disk_index") {
             IndexLimitMb::Unspecified
+        } else {
+            if matches.is_present("disable_accounts_disk_index") {
+                warn!("ignoring `--disable-accounts-disk-index` as it specifies default behavior");
+            }
+            IndexLimitMb::InMemOnly
```

Comment:
_pretty sure_ i got the logic inversion right here.  it's a bit tricky to chase the tri-state through to effective values. seems like it'd be simpler to set a default for `--accounts-index-memory-limit-mb` in the `clap::Arg` def, then just pick between `InMemOnly` and `Limit(limit)` with the `--{en,dis}able-accounts-disk-index` flag